lib/loader.js

```diff
@@ -16,10 +16,14 @@ module.exports = function(content, map) {
 	var root = query.root;
 	var moduleMode = query.modules || query.module;
 	var camelCaseKeys = query.camelCase || query.camelcase;
+	var sourceMap = query.sourceMap || false;
 	var resolve = createResolver(query.alias);
 
-	if(map !== null && typeof map !== "string") {
+	if(sourceMap && map !== null && typeof map !== "string") {
```

Comment:
I think this condition is not correct. If `sourceMap` is true, but the given source map is already a string (which is uncommon in the loader eco-system but can happen as many tools like node-sass return a source map string), it will set `map = null`.

These checks need to be separated I guess.